### DISCUSSION TITLE
CloudWatch Metric Plugin supports graphing Contributor Insight Metrics with INSIGHT_RULE_METRIC() math function

### DISCUSSION THREAD

**Discussion by mnako1** [grafana/grafana#39190] [Data Sources]
CloudWatch Contributor Insights is a feature of CloudWatch which provides streaming TopN analysis of high cardinality time series dimensions.  Customers use CloudWatch Contributor Insights as a cost effective tool for TopN, "Top Talker" and "Most Impact" streaming time series analysis.   

Customer are generally able to graph metrics that describe their Contributor Insight "Rules" using the Math function "INSIGHT_RULE_METRIC(ruleName, metricName)" in an expression.   

This INSIGHT_RULE_METRIC() function is not working in the CloudWatch Grafana datasource plugin however because the "Period" parameter is not being set in the GetMetricData MetricDataQuery object for this expressions.  For most math functions, setting the "Period" parameter on each MetricDataQuery object is not required, however it is required for the "INSIGHT_RULE_METRIC" math function.  

Suggested solution:
  Set the "MetricDataQuery" "Period" Parameter to the Query Period.  

AWS Documentation:  
https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/ContributorInsights-GraphReportData.html
https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_GetMetricData.html
https://docs.aws.amazon.com/AmazonCloudWatch/latest/APIReference/API_MetricDataQuery.html

**Comment by sarahzinger** [comment]
Hello there,

As part of our work to go from [discussions to issues](https://github.com/grafana/grafana/discussions/73424) we're going through some older lingering discussions and seeing which ones we might be able to close and which ones we need to open issues for. If I'm following this correctly, it looks like this feature request was addressed in https://github.com/grafana/grafana/issues/39191 so I'm going to close this discussion. But if I'm missing something feel free to open an issue and we can take another look.

Thanks so much!
